scipy/signal/_fir_filter_design.py

```diff
@@ -872,7 +872,8 @@ def firls(numtaps, bands, desired, weight=None, nyq=None, fs=None):
     bands : array_like
         A monotonic nondecreasing sequence containing the band edges in
         Hz. All elements must be non-negative and less than or equal to
-        the Nyquist frequency given by `nyq`.
+        the Nyquist frequency given by `nyq`. Length of `bands` must be
+        even and strictly greater than zero.
```

Comment:
Can the length of `bands` not be odd as long as it contains pairs? It looks like the source code only requires that the flattened length is even, but the flatted length will always be even if you have an odd collection of pairs?

```python
In [8]: arr = np.array([[0, 0.1], [0.2, 0.3], [0.4, 0.5]])

In [9]: x = scipy.signal.firls(101, arr, [[1, 1], [0, 1], [1, 0]])

In [10]: x
Out[10]: 
array([ 7.86288062e-02, -6.25403763e-01,  2.50544642e+00, -6.58735657e+00,
        1.23360174e+01, -1.66116212e+01,  1.49178528e+01, -5.72374146e+00,
       -5.78985634e+00,  1.01368558e+01, -2.26628623e+00, -1.16697636e+01,
        1.88364203e+01, -1.29990085e+01,  1.82429653e+00,  1.79353271e+00,
        4.79095902e+00, -1.12554910e+01,  8.34777790e+00, -3.71525033e-01,
       -1.30041148e+00, -4.87704302e+00,  7.84261430e+00, -1.67956358e-02,
       -1.02057633e+01,  8.92215643e+00,  3.73903399e+00, -1.31170338e+01,
        9.77422475e+00, -2.02456943e+00,  2.89993824e+00, -1.16462021e+01,
        1.58467977e+01, -9.58362224e+00,  5.15794583e-01,  3.48780591e+00,
       -4.38300179e+00,  7.07157080e+00, -8.62654017e+00,  3.58683750e+00,
        3.61097244e+00, -3.03659129e+00, -5.72091602e+00,  9.85610690e+00,
       -2.26453508e+00, -6.92031055e+00,  5.83787829e+00,  2.71398436e+00,
       -4.45312229e+00, -3.75745060e+00,  1.01209151e+01, -3.75745060e+00,
       -4.45312229e+00,  2.71398436e+00,  5.83787829e+00, -6.92031055e+00,
       -2.26453508e+00,  9.85610690e+00, -5.72091602e+00, -3.03659129e+00,
        3.61097244e+00,  3.58683750e+00, -8.62654017e+00,  7.07157080e+00,
       -4.38300179e+00,  3.48780591e+00,  5.15794583e-01, -9.58362224e+00,
        1.58467977e+01, -1.16462021e+01,  2.89993824e+00, -2.02456943e+00,
        9.77422475e+00, -1.31170338e+01,  3.73903399e+00,  8.92215643e+00,
       -1.02057633e+01, -1.67956358e-02,  7.84261430e+00, -4.87704302e+00,
       -1.30041148e+00, -3.71525033e-01,  8.34777790e+00, -1.12554910e+01,
        4.79095902e+00,  1.79353271e+00,  1.82429653e+00, -1.29990085e+01,
        1.88364203e+01, -1.16697636e+01, -2.26628623e+00,  1.01368558e+01,
       -5.78985634e+00, -5.72374146e+00,  1.49178528e+01, -1.66116212e+01,
        1.23360174e+01, -6.58735657e+00,  2.50544642e+00, -6.25403763e-01,
        7.86288062e-02])
 
In [11]: len(arr)
Out[11]: 3

```

Reply:
Maybe saying the `size` (product of dims) needs to be even? I don't know how much it helps beyond `edges` being inherently pair-related concepts.

Reply:
As you suggest, I think it is important to capture that the bands are paired, and perhaps that the expected input is a 1D array with an even number of elements, or a 2D array consisting of pairs of frequencies (the band edges). For example:

```
bands : array_like
        A monotonic nondecreasing sequence containing the band edges in
        Hz. All elements must be non-negative and less than or equal to
        the Nyquist frequency given by `nyq`. The bands are specified as
        frequency pairs, thus, if using a 1D array, its length must be even,
        e.g., np.array([0, 1, 2, 3, 4, 5]. Alternatively, the bands can be specified
        as an nx2 sized 2D array, where n is the number of bands, e.g,
        np.array([[0, 1], [2, 3], [4, 5]]).
```

Reply:
I think that @RafaelPinto's explanation is way more explicit that just saying "Size of bands must be even". If agreed, I can (or @RafaelPinto can) update the docstring accordingly.

Reply:
@jeertmans @RafaelPinto applying the suggested change makes sense to me I think. If you can do that soon we can probably get this merged in for the `1.10.0` release. Some things to note:

- the first array example is missing a terminating `)` I think: ` np.array([0, 1, 2, 3, 4, 5]`
- you may want to wrap the NumPy arrays in double backticks per convention perhaps? 
- this PR was opened from the main branch on a fork which is usually not advisable--for example, I tried force push the changes I describe but it was denied, probably b/c `main` is protected even on forks
- this is a doc-only change so I think you can adjust the commit message such that only the doc build runs in CI as described here: https://docs.scipy.org/doc/scipy/dev/contributor/continuous_integration.html#skipping

Reply:
As such @tylerjereddy?

Sorry for opening a PR from main, I usually don't do this, my bad.